### PR TITLE
feat(sdk-angular): Improve documentation in @dotcms/angular and angular-example

### LINKED ISSUE #29212: SDK: Improve @dotcms/angular Documentation
### Task description
As a developer using the `@dotcms/angular` library, I want the documentation to be clear and easy to follow, so that I can easily understand how to use the library in my projects.

### Acceptance Criteria
1. The README file should provide a clear introduction to the library and its purpose.
2. The README should include full step by step on how to use the library.
3. Update the README of the `example/angular`, include env variables, tokens, everything needed to start the example
4. Comprehensive JSDoc comments on all the components inputs, properties, classes, methods etc...
6. Any dependencies or prerequisites needed to use the library should be clearly stated.

### Problem
For example the library have this:

```
Add the dotcms config in the Angular app ApplicationConfig

export const appConfig: ApplicationConfig = {
    providers: [
        provideDotcmsClient(DOTCMS_CLIENT_CONFIG),
        provideRouter(routes),
    ],
};
```
 But is not explaining where `provideDotcmsClient` is coming from, or how to create or anything, is missing to much context.

### Assumptions & Initiation Needs
N/A

### PR DESCRIPTION
**Pull request by KevinDavilaDotCMS**
### Proposed Changes
* Removed unused code
* Moved some folders to improve code reading
* Updated README in `@dotcms/angular` and `examples/angular`

Note: I haven't updated the version number of the libs yet, I'll do it before the merge to avoid conflicts

This PR fixes: #29212

### CODE REVIEW

File: examples/angular/src/app/content-types/custom-no-component/custom-no-component.component.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    <div
+      class="relative w-full bg-gray-200 h-12 flex justify-center items-center overflow-hidden"
+    >
+      No component for this "{{contentlet().contentType}}" contentlet or widget
+    </div>
+  `,
+  styles: '',
+})
+export class CustomNoComponent {
+  contentlet = input.required<DotCMSContentlet>();
```
Comment (rjvelazco) [inline examples/angular/src/app/content-types/custom-no-component/custom-no-component.component.ts:17]: Nice!

File: examples/angular/src/app/pages/components/footer/destinations/destinations.component.ts [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
-  InjectionToken,
   OnInit,
   inject,
   signal,
 } from '@angular/core';
 
-import { GenericContentlet } from '..';
-import { ContentletsComponent } from '../contentlets/contentlets.component';
+import { GenericContentlet } from '../..';
+import { ContentletsComponent } from '../../../../shared/contentlets/contentlets.component';
```
Comment (rjvelazco) [inline examples/angular/src/app/pages/components/footer/destinations/destinations.component.ts:9]: Can we create absolute `@paths`?

**Review by rjvelazco — APPROVED** [review]

File: core-web/libs/sdk/angular/README.md
Comment (fmontes) [inline core-web/libs/sdk/angular/README.md:1]: # @dotcms/angular

`@dotcms/angular` is the official Angular library designed to work seamlessly with dotCMS. This library simplifies the process of rendering dotCMS pages and integrating with the [Universal Visual Editor](dotcms.com/docs/latest/universal-visual-editor) in your Angular applications.

## Table of Contents

- [Features](#features)
- [Installation](#installation)
- [Configuration](#configuration)
  - [Provider Setup](#provider-setup)
  - [Client Usage](#client-usage)
- [Components](#components)
  - [DotcmsLayoutComponent](#dotcmslayoutcomponent)
- [Best Practices](#best-practices)
- [Troubleshooting](#troubleshooting)
- [Contributing](#contributing)
- [License](#license)

## Features

- A set of Angular components developer for dotCMS page rendering and editor integration.
- Enhanced development workflow with full TypeScript support.
- Optimized performance for efficient rendering of dotCMS pages in Angular applications.
- Flexible customization options to adapt to various project requirements.

## Installation

Install the package using npm:

```bash
npm install @dotcms/angular
```

Or using Yarn:

```bash
yarn add @dotcms/angular
```

## DotCMS Page API

The `DotcmsLayoutComponent` requires a `DotCMSPageAsset` object to be passed in to it. This object represents a dotCMS page and can be fetched using the `@dotcms/client` library.

- [DotCMS Official Angular Example](https://github.com/dotCMS/core/tree/master/examples/angular)
- [`@dotcms/client` documentation](https://www.npmjs.com/package/@dotcms/client)
- [Page API documentation](https://dotcms.com/docs/latest/page-api)

## Components

### DotcmsLayoutComponent

The `DotcmsLayoutComponent` is a crucial component for rendering dotCMS page layouts in your Angular application.

#### Inputs

| Name         | Type                 | Description                                                           |
|--------------|----------------------|-----------------------------------------------------------------------|
| `pageAsset`  | `DotCMSPageAsset`    | The object representing a dotCMS page from PageAPI response.          |
| `components` | `DotCMSPageComponent`| An object mapping contentlets to their respective render components.  |
| `editor`     | `EditorConfig`       | Configuration for data fetching in Edit Mode.                         |

#### Usage Example

In your component file (e.g., `pages.component.ts`):

```typescript
import { Component, signal } from '@angular/core';
import { DotCMSPageComponent, EditorConfig } from '@dotcms/angular';

@Component({
    selector: 'app-pages',
    templateUrl: './pages.component.html',
})
export class PagesComponent {
    DYNAMIC_COMPONENTS: DotCMSPageComponent = {
        Activity: import('../pages/content-types/activity/activity.component').then(
            (c) => c.ActivityComponent
        ),
        Banner: import('../pages/content-types/banner/banner.component').then(
            (c) => c.BannerComponent
        ),
        // Add other components as needed
    };
    
    components = signal(this.DYNAMIC_COMPONENTS);
    editorConfig = signal<EditorConfig>({ params: { depth: 2 } });
    
    // Assume pageAsset is fetched or provided somehow
    pageAsset: DotCMSPageAsset;
}
```

In your template file (e.g., `pages.component.html`):

```html
<dotcms-layout 
    [pageAsset]="pageAsset" 
    [components]="components()" 
    [editor]="editorConfig()" 
/>
```

This setup allows for dynamic rendering of different content types on your dotCMS pages.

## Best Practices

1. **Lazy Loading**: Use dynamic imports for components to improve initial load times.
2. **Error Handling**: Implement robust error handling for API calls and component rendering.
3. **Type Safety**: Leverage TypeScript's type system to ensure proper usage of dotCMS structures.
4. **Performance Optimization**: Monitor and optimize the performance of rendered components.

## Troubleshooting

If you encounter issues:

1. Ensure all dependencies are correctly installed and up to date.
2. Verify that your dotCMS configuration (URL, auth token, site ID) is correct.
3. Check the browser console for any error messages.
4. Refer to the [dotCMS documentation](https://dotcms.com/docs/) for additional guidance.

Comment (fmontes) [inline core-web/libs/sdk/angular/README.md:1]: The JSDoc in the Angular SDK needs work, methods, interfaces, classes, types... all of those needs to be improved.

**Review by fmontes — CHANGES_REQUESTED** [review]

File: examples/angular/README.md [hunk 3]
```diff
@@ -53,6 +53,30 @@ Once all the configuration is in place, it is time to run the web app.
...
+        │   ├── activity
+        │   ├── banner
+        │   ├── product
+        │   └── ...other-content-types
+        ├── pages/
+        │   ├── components
+        │   └── services
+        └── shared/
+            ├── contentlets
+            └── contentlet
```
Comment (fmontes) [inline examples/angular/README.md:71]: Wayyy to similar naming will confuse people.

**Review by fmontes — CHANGES_REQUESTED** [review]

File: examples/angular/README.md
Comment (fmontes) [inline examples/angular/README.md:1]: # dotCMS Angular Example

This example project demonstrates how to build manageable dotCMS pages headlessly using the Angular framework. It showcases the integration between dotCMS and Angular, providing a practical implementation of content-driven web applications.

## Table of Contents

- [Prerequisites](#prerequisites)
- [Getting Started](#getting-started)
  - [Obtaining the Example Code](#obtaining-the-example-code)
  - [Configuration](#configuration)
- [Running the Application](#running-the-application)
- [Project Structure](#project-structure)
- [Key Features](#key-features)
  - [Handling Vanity URLs](#handling-vanity-urls)
- [Troubleshooting](#troubleshooting)
- [Further Resources](#further-resources)

## Prerequisites

Before you begin, ensure you have the following:

1. Access to a dotCMS instance (you can use https://demo.dotcms.com if you don't have your own)
2. A valid AUTH token for the target dotCMS instance ([How to create an API token](https://auth.dotcms.com/docs/latest/rest-api-authentication#creating-an-api-token-in-the-ui))
3. Node.js (version 18 or higher) and npm installed
4. A terminal application
5. A code editor of your choice

## Getting Started

### Downloading the example

You can get the code in two ways:

1. Direct download:
   ```
   https://github.com/dotCMS/core/tree/master/examples/angular
   ```

2. Using Git sparse checkout:
   ```bash
   git clone -n --depth=1 --filter=tree:0 https://github.com/dotCMS/core
   cd core
   git sparse-checkout set --no-cone examples/angular
   git checkout
   ```
   The example files will be in the `examples/angular` folder.

### Configuration

To configure the Angular app to use your dotCMS instance:

1. Open the project folder in your code editor
2. Navigate to `src/environments`
3. Open `environment.development.ts` and update the following variables:
   - `authToken`: Your dotCMS auth token
   - `dotcmsUrl`: URL of your dotCMS instance (e.g., https://demo.dotcms.com)

   ```typescript
   export const environment = {
     production: false,
     authToken: 'YOUR_AUTH_TOKEN_HERE',
     dotcmsUrl: 'https://demo.dotcms.com',
     // other configurations...
   };
   ```

   ⚠️ **Security Note**: Ensure that the `authToken` used here has read-only permissions to minimize security risks in client-side applications.

## Running the Application

Once configured, follow these steps to run the app:

1. Open a terminal in the project root directory
2. Install dependencies: `npm install`
3. Start the development server: `ng serve`
4. Open your browser and navigate to `http://localhost:4200`

🎉 Congratulations! Your dotCMS Angular example is now running.

Note: When accessing `localhost:4200/about`, ensure that the `/about` page exists in your dotCMS instance.

## Project Structure

```
.
└── src/
    └── app/
        ├── content-types/
        │   ├── activity/
        │   ├── banner/
        │   ├── product/
        │   └── ...other-content-types/
        ├── pages/
        │   ├── components/
        │   └── services/
        └── shared/
            ├── contentlets/
            └── contentlet/
```

- `content-types/`: Components for rendering specific dotCMS content types
- `pages/`: Main application component for rendering pages based on their path and pageAsset
- `shared/`: Reusable components
  - `contentlet/`: Component for rendering individual Contentlets
  - `contentlets/`: Component for displaying lists of Contentlets

## Universal Visual Editor
To enable the Universal Visual Editor in dotCMS, follow these steps:

1. In your dotCMS instance, navigate to the "Apps" page
2. Find the "UVE - Universal Visual Editor" app and click on it
3. Then locate the site where you want to enable the UVE and click on it
4. In the configuration field add the following:

```json
{
  "config": [
    {
      "pattern": ".*",
      "url": "http://localhost:4200"
    }
  ]
}
```

5. Click on the "Save" button to save the changes.
6. Now edit any page and you will see the UVE.

If you want more information about the UVE, please refer to the [dotCMS UVE Documentation](https://dotcms.com/docs/latest/universal-visual-editor-uve).


## Handling Vanity URLs

This example demonstrates how to integrate dotCMS Vanity URLs with Angular routing. Vanity URLs in dotCMS provide alternative paths to internal or external URLs, enhancing site maintenance and SEO.

For implementation details, refer to the [`DotCMSPagesComponent`](./src/app/pages/components/dotcms-pages/dotcms-pages.component.ts) in the example code, which handles routing and Vanity URL redirection.

## Troubleshooting

If you encounter issues:

1. Verify that your dotCMS instance is accessible and the AUTH token is valid
2. Check the browser console for any error messages
3. Ensure all dependencies are correctly installed (`npm install`)
4. Verify that the required pages and content exist in your dotCMS instance

## Further Resources

- [Angular CLI Documentation](https://angular.io/cli)
- [dotCMS Documentation](https://dotcms.com/docs/)
- [dotCMS REST API Authentication](https://auth.dotcms.com/docs/latest/rest-api-authentication)

For more assistance with Angular, use `ng help` or visit the [Angular CLI Overview and Command Reference](https://angular.io/cli) page.

**Review by fmontes — CHANGES_REQUESTED** [review]

File: examples/angular/src/app/pages/components/footer/destinations/destinations.component.ts (hunk 2 above)
Comment (KevinDavilaDotCMS) [reply examples/angular/src/app/pages/components/footer/destinations/destinations.component.ts:9]: I think is not neccesary for the example

File: core-web/libs/sdk/angular/README.md
Comment (KevinDavilaDotCMS) [reply core-web/libs/sdk/angular/README.md:1]: Right, i added jsdoc to a lot of files

**Review by valentinogiardino — APPROVED** [review]

**Review by oidacra — APPROVED** [review]

File: examples/angular/README.md [hunk 4]
```diff
@@ -1,64 +1,153 @@
...
+## Table of Contents
 
-1. A dotCMS instance or you can use https://demo.dotcms.com
-2. A valid AUTH token for the target instance (see: https://auth.dotcms.com/docs/latest/rest-api-authentication#creating-an-api-token-in-the-ui)
-3. Node js 18+ and npm installed
-4. Terminal
-5. And a code editor.
+- [Prerequisites](#prerequisites)
+- [Getting Started](#getting-started)
+  - [Obtaining the Example Code](#obtaining-the-example-code)
```
Comment (fmontes) [inline examples/angular/README.md:9]: Link not working

File: examples/angular/README.md [hunk 5]
```diff
@@ -1,64 +1,153 @@
...
+        │   ├── activity/
+        │   ├── banner/
+        │   ├── product/
+        │   └── ...other-content-types/
+        ├── pages/
+        │   ├── components/
+        │   └── services/
+        └── shared/
+            ├── contentlets/
+            └── contentlet/
```
Comment (fmontes) [inline examples/angular/README.md:98]: These two folder name are too similar, tend to confuse users, we can either rename or change location, or join, or something.

File: examples/angular/README.md [hunk 6]
```diff
@@ -1,64 +1,153 @@
...
+   - `authToken`: Your dotCMS auth token
+   - `dotcmsUrl`: URL of your dotCMS instance (e.g., https://demo.dotcms.com)
 
-Now we need to tell the Angular app what dotCMS instance is going to use to get the data to build its pages.
+   ```typescript
+   export const environment = {
+     production: false,
+     authToken: 'YOUR_AUTH_TOKEN_HERE',
+     dotcmsUrl: 'https://demo.dotcms.com',
+     // other configurations...
```
Comment (fmontes) [inline examples/angular/README.md:63]: What other config can we have here?

File: examples/angular/README.md [hunk 7]
```diff
@@ -1,64 +1,153 @@
...
+   };
+   ```
 
-1. Open the folder `YOUR_NAME` in your code editor
-2. Go to `src/environments`
-3. Open the `environment.development.ts` file and update the environment variable:
+   ⚠️ **Security Note**: Ensure that the `authToken` used here has read-only permissions to minimize security risks in client-side applications.
```
Comment (fmontes) [inline examples/angular/README.md:67]: We need a link here to the permissions docs? Did we have something specific for this? I remeber you wrote something, not sure if Jamie added somewhere.

File: core-web/libs/sdk/angular/README.md [hunk 8]
```diff
@@ -1,17 +1,31 @@
-# @dotcms/angular
++# @dotcms/angular
```
Comment (fmontes) [inline core-web/libs/sdk/angular/README.md:1]: Typo.

File: core-web/libs/sdk/angular/README.md [hunk 9]
```diff
@@ -1,17 +1,31 @@
...
+- [Installation](#installation)
+- [Configuration](#configuration)
+  - [Provider Setup](#provider-setup)
+  - [Client Usage](#client-usage)
+- [Components](#components)
+  - [DotcmsLayoutComponent](#dotcmslayoutcomponent)
+- [Best Practices](#best-practices)
+- [Troubleshooting](#troubleshooting)
+- [Contributing](#contributing)
+- [License](#license)
```
Comment (fmontes) [inline core-web/libs/sdk/angular/README.md:17]: Check the links.

File: core-web/libs/sdk/angular/README.md [hunk 10]
```diff
@@ -23,111 +37,83 @@ Or using Yarn:
...
-    webPageContent: import(
-        '../pages/content-types/web-page-content/web-page-content.component'
-    ).then((c) => c.WebPageContentComponent),
-    Product: import('../pages/content-types/product/product.component').then(
-        (c) => c.ProductComponent,
-    ),
+| Name         | Type                 | Description                                                           |
+|--------------|----------------------|-----------------------------------------------------------------------|
+| `pageAsset`  | `DotCMSPageAsset`    | The object representing a dotCMS page from PageAPI response.          |
+| `components` | `DotCMSPageComponent`| An object mapping contentlets to their respective render components.  |
```
Comment (fmontes) [inline core-web/libs/sdk/angular/README.md:59]: This type is wrong.

File: core-web/libs/sdk/angular/README.md [hunk 11]
```diff
@@ -23,111 +37,83 @@ Or using Yarn:
...
+| `editor`     | `EditorConfig`       | Configuration for data fetching in Edit Mode.                         |
```
Comment (fmontes) [inline core-web/libs/sdk/angular/README.md:60]: Do we have this type?

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/libs/sdk/angular/README.md (hunk 10 above)
Comment (KevinDavilaDotCMS) [reply core-web/libs/sdk/angular/README.md:59]: We have this type and I remember changing it in the lib in this PR, but the change is not there, I already put it again

File: core-web/libs/sdk/angular/README.md (hunk 11 above)
Comment (KevinDavilaDotCMS) [reply core-web/libs/sdk/angular/README.md:116]: Yes!

File: examples/angular/README.md (hunk 6 above)
Comment (KevinDavilaDotCMS) [reply examples/angular/README.md]: No more configs now, removed comment!

**Review by fmontes — APPROVED** [review]

**Pull request merged**